### PR TITLE
docs: k8s auth issuer lookup

### LINKED ISSUE #11953: Support new JWT token structure on k8s 1.21
**Is your feature request related to a problem? Please describe.**
Yes, the current version of Vault (including 1.8) Kubernetes backend does not support Kubernetes clusters on version 1.21.x

**Describe the solution you'd like**
I would like to be able to use Vault with the latest version of the Kubernetes cluster that includes a new more secure JWT token

**Describe alternatives you've considered**
I considered disabling the token claim verification (currently is disabled) or creating a service to rewrite the token in a format that Vault would be able to process 

**Additional context**
Current token format:
```
{
  "iss": "kubernetes/serviceaccount",
  "kubernetes.io/serviceaccount/namespace": "namespace",
  "kubernetes.io/serviceaccount/secret.name": "vault-token-7h4st",
  "kubernetes.io/serviceaccount/service-account.name": "vault",
  "kubernetes.io/serviceaccount/service-account.uid": "6c40a476-8620-869f-a102-6a0d9ad73ef3",
  "sub": "system:serviceaccount:namespace:vault"
}
```

New token format:
```
{
  "aud": [
    "https://container.googleapis.com/v1/projects/project-name/locations/europe-west1/clusters/cluster-name"
  ],
  "exp": 1656040272,
  "iat": 1624504272,
  "iss": "https://container.googleapis.com/v1/projects/project-name/locations/europe-west1/clusters/cluster-name",
  "kubernetes.io": {
    "namespace": "namespace",
    "pod": {
      "name": "tmp",
      "uid": "cc2866c7-fee8-5613-a563-15613d152ee2"
    },
    "serviceaccount": {
      "name": "vault",
      "uid": "6c40a476-8620-869f-a102-6a0d9ad73ef3"
    },
    "warnafter": 1624507879
  },
  "nbf": 1624504272,
  "sub": "system:serviceaccount:namespace:vault"
}
```

### PR DESCRIPTION
**Pull request by tvoran**
Link to the instructions for looking up a Kubernetes cluster's service
account issuer.

Closes #11953

### CODE REVIEW

**Review by tomhjp — COMMENTED** [review]
Thanks for opening this. I think maybe we should move the instructions from CSI to k8s auth though, and then link the other way round? The issue isn't really specific to CSI in any way. WDYT?

**Comment by tvoran** [comment]
@tomhjp Yeah, that makes more sense, I'll try moving the actual instructions to the k8s auth section.

**Comment by tvoran** [comment]
@tomhjp Moved the instructions from CSI to k8s auth in 543a70730

File: website/content/docs/auth/kubernetes.mdx [hunk 1]
```diff
@@ -99,8 +98,47 @@ management tool.
...
+  | cut -d. -f2 \
+  | base64 -D
+```
+
+Most clusters will also have that information available at the
+`.well-known/openid-configuration` endpoint:
+
+```bash
+kubectl proxy &
+curl --silent http://127.0.0.1:8001/.well-known/openid-configuration | jq -r .issuer
```
Comment (tomhjp) [inline website/content/docs/auth/kubernetes.mdx:130]: Nice addition!

**Review by tomhjp — APPROVED** [review]
Looks great 👌 thanks for sorting this.

**Pull request merged**